lib/client/file-ng.js

```diff
@@ -0,0 +1,44 @@
+'use strict';
+
+var client = require('./file'),
+  _ = require('lodash');
+
+const MODULE_NAME = 'wfm.file.module';
+var defaultConf = require("./web-config");
+
+
+/**
+ * Initialize Raincatcher file module
+ *
+ * @param userConfig - object containing user overrides for default module configuration
+ * See ./web-config
+ *
+ * @returns {string} module name
+ */
+module.exports = function(userConfig){
+  userConfig = userConfig ? userConfig : {};
+  _.defaults(userConfig, defaultConf);
+  var directivesModule = require('./directives/index');
+  var mainModule = angular.module(MODULE_NAME, [
+    directivesModule
+  ]);
+  
+  // TODO Remove factory and use mediator
+  mainModule.factory('fileClient', function($q){
```

Comment:
Going to work on this. Posting this PR for early feedback and also to provide template for refactoring other modules.